### DISCUSSION TITLE
How to resolve merge conflicts after removing a file from LFS?

### DISCUSSION THREAD

**Discussion by sisoor** [git-lfs/git-lfs#4583] [Q&A]
I have two branches:
1. "upstream": Tracking a file extension, then a single file with this extension is moved out of LFS.
2. "downstream": Created from "upstream" before untracking that file. Also has changes to that file.

Now when I merge "upstream" into "downstream" I get a merge conflict in the file in question with: original = LFS pointer, ours = LFS pointer, theirs = file content.

```
git init
git checkout -b upstream
echo '**/*.xml filter=lfs merge=lfs diff=lfs text' > .gitattributes
echo foo > foo.xml
git add .
git commit -m 'initial commit'
echo '**/foo.xml !filter !merge !diff text' >> .gitattributes
git add --renormalize .
git commit -m 'remove foo.xml from lfs'
git checkout -b downstream upstream^
echo bar >> foo.xml
git add .
git commit -m 'extend foo'
git merge --no-ff upstream
```
How would I resolve this conflict? Is my change to the .gitattributes wrong?

Using git version 2.31.1 with git-lfs/2.13.3.

**Comment by chrisd8088** [comment]
Good question!  I suspect the likely thing you want to do here would be recreate `foo.xml` as a regular file rather than a Git LFS object, but with the changes you'd made on your `downstream` branch in its contents.

So at the point where your example stops, I might do:
```
$ git merge upstream
Auto-merging foo.xml
CONFLICT (content): Merge conflict in foo.xml
Automatic merge failed; fix conflicts and then commit the result.

$ git show HEAD:foo.xml | git lfs smudge >foo.xml
$ cat foo.xml
foo
bar

$ git add foo.xml
$ git commit -m 'merge upstream and remove foo.xml from lfs'

$ git show HEAD:foo.xml
foo
bar
```

So the merge commit will include the change from the Git LFS pointer file to a regular file, one with the contents created by the most recent commit on the `downstream` branch before the merge.  I hope that helps a bit?

**Reply by sisoor** [reply]
Thanks for your answer! This would work for my example but would discard upstream changes if there were any - something I cannot rule out unfortunately.
